openni2_camera/src/openni2_driver.cpp

```diff
@@ -245,7 +245,15 @@ void OpenNI2Driver::setColorVideoMode(const OpenNI2VideoMode& color_video_mode)
   {
     if (color_video_mode != device_->getColorVideoMode())
     {
+      if (device_->isColorStreamStarted())
+      {
+        device_->stopColorStream();
+      }
       device_->setColorVideoMode(color_video_mode);
+      if (pub_color_.getNumSubscribers() > 0)
```

Comment:
For my understanding, this check on the number of subscribers has the purpose of:
- A small optimization.
- Removes warning messages.
- Prevents an actual error.

Reply:
The original code uses this check to start the colorStream, as an optimisation. See [L](https://github.com/Intermodalics/openni2_camera/blob/ros1/openni2_camera/src/openni2_driver.cpp#L386)